src/Infrastructure/BotSharp.MCP/Functions/McpToolFunction.cs

```diff
@@ -0,0 +1,106 @@
+using BotSharp.Abstraction.Agents;
+using BotSharp.Abstraction.Conversations.Models;
+using BotSharp.Abstraction.Functions;
+using BotSharp.Abstraction.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+using ModelContextProtocol.Client;
+using ModelContextProtocol.Protocol.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace BotSharp.Core.Mcp.Functions;
+
+public class McpToolFunction : IFunctionCallback
```

Comment:
Do you think the name `McpToolAdapter` is better?

Reply:
It is primarily responsible for converting calls or responses from the MCP tools to fit the API expected by BotSharp, “Adapter” would immediately signal this role to other developers. It interacts with mcp server  and converts data between incompatible mcp  tools interfaces, then McpToolAdapter is a clearer, more descriptive choice